File: vargplvm/matlab/kernVardistPsi0Compute.m
function Psi0 = kernVardistPsi0Compute(kern, vardist)

% KERNVARDISTPSI0COMPUTE description.  

% VARGPLVM


if ~strcmp(kern.type,'cmpnd')
  %  
  fhandle = str2func([kern.type 'VardistPsi0Compute']);
  Psi0 = fhandle(kern, vardist);
  %
else  % the kernel is cmpnd
  % 
  fhandle = str2func([kern.comp{1}.type 'VardistPsi0Compute']);
  Psi0 = fhandle(kern.comp{1}, vardist);
  %
  for i = 2:length(kern.comp)
      %
      fhandle = str2func([kern.comp{i}.type 'VardistPsi0Compute']);
      Psi0 = Psi0 + fhandle(kern.comp{i}, vardist);
      %
  end
  %
end

File: vargplvmR/R/kernVardistPsi0Compute.R
kernVardistPsi0Compute <-
function (kern, vardist)
{
# % KERNVARDISTPSI0COMPUTE description not available.  
# % FORMAT
# % DESC 
# % description not available

if (!(kern$type == "cmpnd"))
{
  fhandle <- paste(kern$type, "VardistPsi0Compute",sep="")
  Psi0 <- do.call(fhandle,list(kern, vardist))
}
else  #% the kernel is cmpnd
{
  fhandle <- paste(kern$comp[[1]]$type, "VardistPsi0Compute", sep ="")
  Psi0 <- do.call(fhandle, list(kern$comp[[1]], vardist))
  
  for (i in 2:length(kern$comp))
  {
      fhandle <- paste(kern$comp[[i]]$type, "VardistPsi0Compute", sep = "")
      Psi0 <- Psi0 + do.call(fhandle, list(kern$comp[[i]], vardist))
  }
}

return (Psi0)
}
